### PR TITLE
added a pubsub-chat example

### PR DESCRIPTION
**Pull request by bennewhall**
I added a pubsub example based on the libp2p-chat example.

Usage is described in the comments.

Please let me know if there is anything I should change!

### CODE REVIEW

File: example/pub-sub-chat.ss [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+      (begin
+        (displayln "Bootsrap peer found, connecting to it")
+        (libp2p-connect c peer)
+        (displayln "Connected to Bootstrap peer")))
+
+    ;Subscribe to the chat and spawn a reader thread and writer thread
+    (let*-values (((sub cancel) (pubsub-subscribe c "chat"))
+           (reader (spawn subscription-reader sub self)))
+      (subscription-writer c)
+      (thread-terminate! reader)))) ;;TODO: Does this exit cleanly?
```
Comment (vyzo) [inline example/pub-sub-chat.ss:113]: no, it really abruptly kills the thread -- you might want to raise an exception in it to interrupt it. See `:std/misc/threads`, you can use `thread-raise!` or `thread-abort!`.

**Review by vyzo — COMMENTED** [review]
looks fine, and a very welcome contribution. Only (nitpicky) suggestion I would make is to call the program `pubsub-chat` instead of `pub-sub-chat`.

File: example/pub-sub-chat.ss (hunk 1 above)
Comment (bennewhall) [reply example/pub-sub-chat.ss:113]: Sounds good! Is there a way to make sure the daemon exits cleanly as well? Or will this do it?

Comment (bennewhall) [reply example/pub-sub-chat.ss]: Just pushed your suggestions!

**Review by vyzo — APPROVED** [review]

File: example/pub-sub-chat.ss (hunk 1 above)
Comment (vyzo) [reply example/pub-sub-chat.ss]: i think it should be fine, but might make sense to explicitly close the client somewhere.

**Pull request merged**